src/model/channel/channel_id.rs

```diff
@@ -19,6 +19,8 @@ use crate::utils;
 use crate::http::{Http, CacheHttp};
 #[cfg(feature = "model")]
 use serde_json::json;
+#[cfg(feature = "cache")]
```

Comment:
The `MessagesIter` stream doesn't utilize the cache at all. This just unnecessarily restricts usage in contexts where the cache is disabled.
```suggestion

```